### ISSUE TITLE
Manifest repo locations cannot be overridden

### ISSUE THREAD

**Issue by dborman-hpe** [Cray-HPE/loftsman#40]
#### What happened?
Previously we were downloading all our helm charts from one location, and then running loftsman pointing at the local directory, i.e.:
```
loftsman ship --charts-path /opt/cray/helm_charts --manifest-path build/manifests/platform.yaml
```
Due to the need to download helm charts from multiple locations, the manifests were updated with a `sources.charts:` section, and a `source:` was added to to each chart.  With this change, we are able to download the charts from multiple locations.

However, when I then used loftsman to run the manifest as shown above, the `--charts-path` information was overridden by the `source:` specified in the manifest, leading to failure:
```
2021-09-08T12:00:06-07:00 INF Ensuring that the loftsman namespace exists command=ship
2021-09-08T12:00:06-07:00 INF Loftsman will use the packaged charts at /opt/cray/helm_charts as the Helm install source command=ship
2021-09-08T12:00:06-07:00 INF Running a release for the provided manifest at build/manifests/platform.yaml command=ship

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Releasing cray-metrics-server v0.1.2
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

2021-09-08T12:00:07-07:00 INF Running helm install/upgrade with arguments: upgrade --install cray-metrics-server https://arti.dev.cray.com/artifactory/csm-helm-stable-local/https:/arti-web-lb.dev.cray.com/artifactory/csm-helm-stable-local/cray-metrics-server-0.1.2.tgz --namespace kube-system --create-namespace --set global.chart.name=cray-metrics-server --set global.chart.version=0.1.2 chart=cray-metrics-server command=ship namespace=kube-system version=0.1.2
2021-09-08T12:00:07-07:00 ERR Error releasing chart cray-metrics-server v0.1.2: Shell error: Release "cray-metrics-server" does not exist. Installing it now.
Error: failed to download "https://arti.dev.cray.com/artifactory/csm-helm-stable-local/https:/arti-web-lb.dev.cray.com/artifactory/csm-helm-stable-local/cray-metrics-server-0.1.2.tgz" (hint: running `helm repo update` may help) chart=cray-metrics-server command=ship namespace=kube-system version=0.1.2
```
#### What did you expect to happen?
I expected the command line specification of `--charts-path` to override the information in the manifests

#### How to reproduce the issue (as minimally and precisely as possible):
Add a `spec.sources` section to a manifest, add a `source:` entry to the `spec.charts` entries, download the chart to a local directory, and then run `loftsman` with `--charts-path <local_directory>`

#### Anything else worth sharing:
It appears that currently the only way around this issue is to create a second manifest that changes the `spec.sources` to be of type directory and the location of the directory to which the helm charts were downloaded.

**Comment by rockholla** [comment]
Hey David, first, just verify the version of loftsman being used in this case. Second, if you could share the manifest itself. Feel free to omit/obscure `values` overrides, but I do want to see how `spec.sources` are set and how the charts are set to point to it.

`https://arti.dev.cray.com/artifactory/csm-helm-stable-local/https:/arti-web-lb.dev.cray.com/artifactory/csm-helm-stable-local/cray-metrics-server-0.1.2.tgz` looks like [a bug that was fixed](https://github.com/Cray-HPE/loftsman/commit/d6099fc4443ff3655b169a9a60beef8a423f4639) in relation to chart repos that serve up a full URL path in their index.yaml entries, so I suspect you might not be on [loftsman version 1.1.1](https://github.com/Cray-HPE/loftsman/releases/tag/1.1.1) which contains this fix.

Last thing is that `--charts-path` is kept around for backwards compatibility, but you can stop using it in the `loftsman` binary call when you move to using `spec.sources`.

**Comment by dborman-hpe** [comment]
I'm using loftsman 1.1.0.  What I am doing now to deal with this is to re-write the manifest and change the spec.sources to be of type directory with the location for the local directory where the helm charts were downloaded.

The workflow that is being used is to first use the manifest to download all the helm charts, and then use the manifest with "loftsman ship" to deploy the manifests, but instead of using the download locations specified in the sources, we want them deployed using the directory to which the charts were downloaded.  In the past the `--charts-path` achieved that second step.

This is all in support of deploying into an air-gapped environment.  During the build of the ISO image we download all the helm charts (and containers), and then at runtime we use the local copies.

**Comment by rockholla** [comment]
If you can share the manifest, I can suggest the changes you need to it with that workflow

**Comment by dborman-hpe** [comment]
An abbreviated manifest is:

**Comment by dborman-hpe** [comment]
An abbreviated manifest is:
```
apiVersion: manifests/v1beta1
metadata:
  name: base
spec:
  sources:
    charts:
    - name: remote
      type: repo
      location: https://foo.org/helm/
  charts:
  - name:metrics-server
    source: remote
    version: 0.1.2
    namespace: kube-system
```

**Comment by rockholla** [comment]
This shared manifest is OK. The initial error shared is a bug that was fixed in 1.1.1, so I'd suggest getting that version as your primary fix for this issue filed. For detailed info on what you can do with remote and local chart sources as defined in the manifest, I'd suggest looking at the [comprehensive manifest example in this repo](https://github.com/Cray-HPE/loftsman/blob/main/schemas/manifests/v1beta1/examples/comprehensive.yaml).

Again, you can/should stop using the `charts-path` arg and instead rely on the `spec.sources` only.

I'm happy to point you in the right direction for anything else you need, but closing this issue for now.

**Comment by dborman-hpe** [comment]
I'm working around this by adding scripting that takes the manifest and modifies the `spec.sources` section to point to the local directory, and then passing that modified manifest to loftsman.
